src/mlpack/methods/linear_svm/linear_svm_main.cpp

```diff
@@ -0,0 +1,419 @@
+PARAM_INT_IN("number_of_classes", "Number of classes for classification; if "
+    "unspecified (or 0), the number of classes found in the labels will be "
+    "used.", "c", 0);
+PARAM_FLAG("no_intercept", "Do not add the intercept term to the model.", "N");
+PARAM_STRING_IN("optimizer", "Optimizer to use for training ('lbfgs' or "
+    "'psgd').", "O", "lbfgs");
+PARAM_DOUBLE_IN("tolerance", "Convergence tolerance for optimizer.", "e",
+    1e-10);
+PARAM_INT_IN("max_iterations", "Maximum iterations for optimizer (0 indicates "
+    "no limit).", "n", 10000);
+PARAM_DOUBLE_IN("step_size", "Step size for parallel SGD optimizer.",
+    "s", 0.01);
+PARAM_FLAG("shuffle", "Don't shuffle the order in which data points are "
+    "visited for parallel SGD.", "S");
+PARAM_INT_IN("epochs", "Maximum number of full epochs over dataset for "
+    "psgd", "E", 50);
+PARAM_INT_IN("seed", "Random seed.  If 0, 'std::time(NULL)' is used.", "r", 0);
+
+// Model loading/saving.
+PARAM_MODEL_IN(LinearSVM<>, "input_model", "Existing model "
+    "(parameters).", "m");
+PARAM_MODEL_OUT(LinearSVM<>, "output_model", "Output for trained "
+    "linear svm model.", "M");
+
+// Testing.
+PARAM_MATRIX_IN("test", "Matrix containing test dataset.", "T");
+PARAM_UROW_IN("test_labels", "Matrix containing test labels.", "A");
+PARAM_UROW_OUT("predictions", "If test data is specified, this matrix is where "
+    "the predictions for the test set will be saved.", "P");
+PARAM_MATRIX_OUT("score", "If test data is specified, this "
+    "matrix is where the class score for the test set will be saved.",
+    "p");
+
+static void mlpackMain()
+{
+  if (CLI::GetParam<int>("seed") != 0)
+    math::RandomSeed((size_t) CLI::GetParam<int>("seed"));
+  else
+    math::RandomSeed((size_t) std::time(NULL));
+
+  // Collect command-line options.
+  const double lambda = CLI::GetParam<double>("lambda");
+  const double delta = CLI::GetParam<double>("delta");
+  const string optimizerType = CLI::GetParam<string>("optimizer");
+  const double tolerance = CLI::GetParam<double>("tolerance");
+  const bool intercept = CLI::HasParam("no_intercept");
+  const size_t epochs = (size_t) CLI::GetParam<int>("epochs");
+  const size_t maxIterations = (size_t) CLI::GetParam<int>("max_iterations");
+
+  // One of training and input_model must be specified.
+  RequireAtLeastOnePassed({ "training", "input_model" }, true);
+
+  // If no output file is given, the user should know that the model will not be
+  // saved, but only if a model is being trained.
+  RequireAtLeastOnePassed({ "output_model", "predictions", "score"},
+      false, "no output will be saved");
+
+  ReportIgnoredParam({{ "test", false }}, "predictions");
+  ReportIgnoredParam({{ "test", false }}, "score");
+  ReportIgnoredParam({{ "test", false }}, "test_labels");
+
+  // Max Iterations needs to be positive.
+  RequireParamValue<int>("max_iterations", [](int x) { return x >= 0; },
+      true, "max_iterations must be positive or zero");
```

Comment:
We can just use `non-negative` in place of `positive of zero`. :+1:

Reply:
Updated.